src/tracing/tracing.c

```diff
@@ -120,8 +131,15 @@ static void xdebug_stop_trace(void)
 		return;
 	}
 
-	XG_TRACE(trace_handler)->write_footer(XG_TRACE(trace_context));
-	XG_TRACE(trace_handler)->deinit(XG_TRACE(trace_context));
+	if (XG_TRACE(trace_handler)->write_footer) {
+		XG_TRACE(trace_handler)->write_footer(XG_TRACE(trace_context));
+	}
+	if (XG_TRACE(trace_handler)->deinit) {
+		XG_TRACE(trace_handler)->deinit(XG_TRACE(trace_context));
+	} else {
+		/* @todo there should an explicit error here to expose to the handler developer
+		   that a missing deinit handlers means a memory leak. */
```

Comment:
Same here, having a memory leak on trace shutdown is probably not a problem, since it's not going to happen on any production, but an explicit warning in the console would greatly help during handler development.